addons/crm/models/crm_lead.py

```diff
@@ -1706,7 +1706,7 @@ def _find_matching_partner(self, email_only=False):
             # search through the existing partners based on the lead's partner or contact name
             # to be aligned with _create_customer, search on lead's name as last possibility
             for customer_potential_name in [self[field_name] for field_name in ['partner_name', 'contact_name', 'name'] if self[field_name]]:
-                partner = self.env['res.partner'].search([('name', 'ilike', '%' + customer_potential_name + '%')], limit=1)
+                partner = self.env['res.partner'].search([('name', 'ilike', f'%{customer_potential_name}%')], limit=1)
```

Comment:
https://www.odoo.com/documentation/master/developer/reference/backend/orm.html#search-domains

like
matches field_name against the %value% pattern. Similar to =like but wraps value with ‘%’ before matching

so the % is not need here.